test/Microsoft.AspNet.Routing.Tests/Template/TemplateParserTests.cs

```diff
@@ -634,10 +688,7 @@ public void InvalidTemplate_InvalidParameterNameWithOpenBracketThrows()
         {
             ExceptionAssert.Throws<ArgumentException>(
                 () => TemplateParser.Parse("{a}/{a{aa}/{z}"),
```

Comment:
Please include a **valid** case like this `{p{{}` (parameter name = `p{`). Is this a new thing that would not have been valid before.

Also `{p/}` or similar